actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -57,6 +57,8 @@ def self.empty
       new({})
     end
 
+    attr_accessor :route_uri_pattern
```

Comment:
We should implement this using the rack env so middleware can also have access to the information. See `routes=` for example.